### PR TITLE
[DOC] Glossary rewrite

### LINKED ISSUE #361: [DOC] Link definitions in glossary page
### Describe the issue linked to the documentation

Once #346 is in, we should re-visit the linking between definitions to make the flow of reading easier for the users. For instance when defining a `Multivariate Time Series`, the text should link to the definition of `Time Series` given its way below it because of the sorting attribute.

### Suggest a potential alternative/fix

_No response_

### LINKED ISSUE #601: [DOC] Glossary of terms
### Describe the issue linked to the documentation

a like the idea of a glossary
https://aeon-toolkit--598.org.readthedocs.build/en/598/glossary.html
but it needs updating

### Suggest a potential alternative/fix

Update/edit/remove

### PR DESCRIPTION
**Pull request by MatthewMiddlehurst**
#### Reference Issues/PRs

Resolves #601
Resolves #361  

#### What does this implement/fix? Explain your changes.

Rewrites the glossary page mostly. Removes todos and some previous used terminology.

Some sections are the same, others are completely different.

#### What should a reviewer concentrate their feedback on?

The new glossary entries.

### CODE REVIEW

File: docs/glossary.md [hunk 1]
```diff
@@ -1,104 +1,155 @@
...
-The glossary below defines common terms and API elements used throughout ``aeon``.
+The glossary below defines common terms and API elements used throughout `aeon`.
 
 ```{glossary}
 :sorted:
 
+Time series data
+Time series
+Series
+    Data with multiple individual {term}`variable` measurements with acoompanying
```
Comment (TonyBagnall) [inline docs/glossary.md:11]: accompanying

File: docs/glossary.md [hunk 2]
```diff
@@ -1,104 +1,155 @@
...
+    (e.g. time-invariant measurements like a patient's place of birth) or a sequence
+    of multiple values as a {term}`time series`.
+
+    For time series data, multiple variables may be refered to as {term}`channels`.
+
+Target variable
+Target variables
+    The {term}`variable`(s) that are being predicted in a learning task using
+    {term}`Independant variables`, past {term}`timepoints` of the variable itself, or
+    both. Also referred toas the dependent or endogenous variable(s).
```
Comment (TonyBagnall) [inline docs/glossary.md:34]: toas -> to as
this does not scan very well

File: docs/glossary.md [hunk 3]
```diff
@@ -1,104 +1,155 @@
...
+
+Independant variable
+Independant variables
```
Comment (TonyBagnall) [inline docs/glossary.md:37]: independent

File: docs/glossary.md [hunk 4]
```diff
@@ -1,104 +1,155 @@
...
+Instance
+    A member of the set of entities being studied and which an machine learning
+    practitioner wishes to generalize. For example, patients, chemical process runs,
+    machines, countries, etc.
+
+    May also be referred to as cases, samples, examples, observations or records
+    depending on the discipline and context.
 
+Panel
 Panel time series
```
Comment (TonyBagnall) [inline docs/glossary.md:122]: Panel time series -> Time series panel (?)

File: docs/glossary.md [hunk 5]
```diff
@@ -1,104 +1,155 @@
...
 
 Time series collection
-    todo
+Time series collections
+    A datatype which contains multiple {term}`instances` of time series. These series
+    may be {term}`univariate time series` or {term}`multivariate time series`. The time
+    series contained within may be of different lengths, sampled at different
+    frequencies, contain differing {term}`timepoints` etc.
 
+    Commonly referred to as a {term}`panel time series` depending on context and
```
Comment (TonyBagnall) [inline docs/glossary.md:132]: Commonly referred to -> Also referred to

File: docs/glossary.md [hunk 6]
```diff
@@ -1,104 +1,155 @@
...
+    A single {term}`time series`.
 
+Multivariate
 Multivariate time series
-    Multiple time series. Typically observed for the same observational unit.
-    Multivariate time series is typically used to refer to cases where the series evolve
-    together over time. This is related, but different than the cases where a
-    {term}`univariate time series` is dependent on {term}`exogenous` data.
-
-Endogenous
```
Comment (TonyBagnall) [inline docs/glossary.md:68]: Endogenous and Exogenous are common terms in forecasting

File: docs/glossary.md [hunk 7]
```diff
@@ -1,104 +1,155 @@
...
-Exogenous
-    Within a learning task exogenous variables are external factors whose pattern of
-    impact on tasks' endogenous variables must be learned.
-    Also referred to as independent variables or features.
+    An example of time series data with multiple channels is data extracted from a
+    gyroscope sensor, which can produce different time series data for the x, y and
+    z axes of the device.
 
 Reduction
     Reduction refers to decomposing a given learning task into simpler tasks that can
```
Comment (TonyBagnall) [inline docs/glossary.md:150]: does this sentence end with can? May jsut be formatting view

File: docs/glossary.md [hunk 8]
```diff
@@ -1,104 +1,155 @@
...
-Instance
-    A member of the set of entities being studied and which an ML practitioner wishes
-    to generalize. For example, patients, chemical process runs, machines, countries,
-    etc. May also be referred to as samples, examples, observations or records
-    depending on the discipline and context.
+    be composed to create a solution to the original task. In `aeon` reduction is used
+    to allow one learning task to be adapted as a solution for an alternative task.
 
 Trend
     When data shows a long-term increase or decrease, this is referred to as a trend.
```
Comment (TonyBagnall) [inline docs/glossary.md:155]: When time series show

**Review by TonyBagnall — CHANGES_REQUESTED** [review]

File: docs/glossary.md [hunk 9]
```diff
@@ -1,104 +1,155 @@
...
 
+Time series data
+Time series
+Series
+    Data with multiple individual {term}`variable` measurements with acoompanying
+    {term}`timepoints` which are ordered over time or have an index indicating the
+    position of an observation in the sequence of values.
+
+Timepoint
+Timepoints
```
Comment (TonyBagnall) [inline docs/glossary.md:16]: do we really need singular and plural listed on all of these?

File: docs/glossary.md (hunk 7 above)
Comment (MatthewMiddlehurst) [reply docs/glossary.md:150]: Reduction
    Reduction refers to decomposing a given learning task into simpler tasks that can
    be composed to create a solution to the original task. In `aeon` reduction is used
    to allow one learning task to be adapted as a solution for an alternative task.

File: docs/glossary.md (hunk 9 above)
Comment (MatthewMiddlehurst) [reply docs/glossary.md:16]: I imagine it helps with searching a referencing, i.e. allowing both {term}`Instance` and {term}`Instances` to link to it in the docs.

File: docs/glossary.md (hunk 6 above)
Comment (TonyBagnall) [reply docs/glossary.md:68]: hmmm, seems a bit more complex than that, seems endogenous means correlated with the error and exogenous means uncorrelated, although of course error is only estimated. I dont mind ducking it

Comment (MatthewMiddlehurst) [reply docs/glossary.md:68]: I'm not sure to be honest. I dont think the old ones were that much better in this regard though.

**Review by TonyBagnall — APPROVED** [review]

**Pull request merged**